packages/cdktf-cli/bin/cmds/helper/synth-stack.ts

```diff
@@ -57,28 +69,28 @@ Command output on stderr:
     await this.synthTelemetry(command, (endTime - startTime));
 
     const stacks: SynthesizedStack[] = [];
-
-    for (const file of await fs.readdir(outdir)) {
-      if (file.endsWith('.tf.json')) {
-        const filePath = path.join(outdir, file);
-        const jsonContent: SynthesizedStackMetadata = JSON.parse(fs.readFileSync(filePath).toString());
-        const name = jsonContent['//']?.metadata.stackName;
-        if (name !== undefined) {
-          stacks.push({
-            file: path.join(outdir, file),
-            name,
-            content: JSON.stringify(jsonContent, null, 2)
-          })
-        }
-      }
+    const manifest = JSON.parse(fs.readFileSync(path.join(outdir, Manifest.fileName)).toString()) as ManifestJson
+
+    for (const stackName in manifest.stacks) {
+      const stack = manifest.stacks[stackName];
+      const filePath = path.join(outdir, stack.synthesizedStackPath);
+      const jsonContent: SynthesizedStackMetadata = JSON.parse(fs.readFileSync(filePath).toString());
+      stacks.push({
+        ...stack,
+        workingDirectory: path.join(outdir, stack.workingDirectory),
+        content: JSON.stringify(jsonContent, null, 2)
+      })
     }
 
     if (stacks.length === 0) {
       console.error('ERROR: No Terraform code synthesized.');
     }
 
-    if (stacks.length > 1) {
-      console.error('ERROR: Found more than one stack. Multiple stacks are not supported at the moment and might lead to unpredictable behaviour.');
+    const stackNames = stacks.map(s => s.name)
+    const orphanedDirectories = existingDirectories.filter(e => !stackNames.includes(path.basename(e)))
+
+    for (const orphanedDirectory of orphanedDirectories) {
+      fs.rmdirSync(orphanedDirectory, { recursive: true })
```

Comment:
Could this somehow lead to unexpected side effects like loosing local Terraform state after renaming a stack?

Reply:
Just checked. The local Terraform state is writting to the root directory and not touched 👍